### PR TITLE
rosbag2_transport: parametrize test_rewrite

### PR DESCRIPTION
**Pull request by james-rms**
runs `test_rewrite` using all tested storage IDs, for better test coverage.

Signed-off-by: James Smith <james@foxglove.dev>

### CODE REVIEW

**Comment by james-rms** [comment]
Gist: https://gist.githubusercontent.com/james-rms/782d820aa5fac60a9c4b27124b678895/raw/53f6d77935969c69bced3123482fb66971c53af3/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_transport
TEST args:  --packages-above rosbag2_transport
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/11257
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17801/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12364/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18457/)

File: rosbag2_transport/package.xml [hunk 1]
```diff
@@ -30,6 +30,7 @@
   <test_depend>rosbag2_compression_zstd</test_depend>
   <test_depend>rosbag2_test_common</test_depend>
   <test_depend>rosbag2_storage_default_plugins</test_depend>
+  <test_depend>rosbag2_storage_mcap</test_depend>
```
Comment (emersonknapp) [inline rosbag2_transport/package.xml:33]: Still feeling a bit awkward about this, is there a very strong reason we shouldn't add mcap to the `exec_depend` of default plugins? We are planning to ship it, so it doesn't seem to me that there's any harm in doing it.

Or, if we are trying to be very explicit about what we're testing, then maybe `rosbag2_storage_sqlite3` is a `test_depend` here, and `default_plugins` is not.

**Review by emersonknapp — APPROVED** [review]
One non-blocking thought - depending how you respond maybe it needs a `TODO` at least.

Also, I have been thinking that maybe all of these test bags should be generated in code, rather than checked in to the repo. That way it would be easier to just iterate the storage plugins (or whatever other criteria) and test everything, without manual asset creation. The one place where I see actual checked-in file artifacts being useful is for old format backwards compatibility. But, that could be a really explicit list that has some standard battery run on them.

File: rosbag2_transport/package.xml (hunk 1 above)
Comment (james-rms) [reply rosbag2_transport/package.xml:33]: I'm trying to merge these test coverage PRs with the least externally visible disruption possible, which means not adding `rosbag2_storage_mcap` as a default plugin until it actually is one. That might mean a couple of extra lines like this one in some package.xml files which will be reverted later, but I don't think that's too bad.

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio humble

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyiobackport humble

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio  backport humble

**Pull request merged**